packages/sdk/src/__tests__/integration/modules/middlewareModule.spec.ts

```diff
@@ -491,4 +495,72 @@ describe("middlewareModule", () => {
 
     expect(res).toEqual({ id: 1, name: "Custom method" });
   });
+
+  it("should throw an error as an instance of SdkHttpError class with default http client", async () => {
+    expect.assertions(6);
+    const sdk = initSDK({
+      commerce: buildModule(middlewareModule<Endpoints>, {
+        apiUrl: "http://localhost:8181/commerce",
+      }),
+    });
+
+    try {
+      // This is a real request to the middleware, invalid endpoint throws { statusCode: 401, message: "Unauthorized" }
+      await sdk.commerce.unauthorized();
+    } catch (err) {
+      expect(err).toBeInstanceOf(Error);
+      expect(isCausedBySdkHttpError(err)).toBe(true);
+      expect(isSpecificSdkHttpError(err, { statusCode: 401 })).toBe(true);
+      expect(
+        isSpecificSdkHttpError(err, {
+          statusCode: (statusCode) => statusCode === 401,
+        })
+      ).toBe(true);
+      expect(isSdkRequestError(err)).toBe(true);
+      expect(isSdkUnauthorizedError(err)).toBe(true);
+    }
+  });
+
+  it("should throw an error as an instance of SdkHttpError class with axios as a custom http client", async () => {
+    expect.assertions(6);
+
+    const sdk = initSDK({
+      commerce: buildModule(middlewareModule<Endpoints>, {
+        apiUrl: "http://localhost:8181/commerce",
+        httpClient: async (url, params, config) => {
+          try {
+            const { data } = await axios(url, {
+              ...config,
+              data: params,
+              withCredentials: true,
+            });
+
+            return data;
+          } catch (err: any) {
+            throw new SdkHttpError({
+              statusCode: err?.response?.status || 500,
+              message: err?.response?.data?.message || err.message,
+              cause: err,
+            });
+          }
```

Comment:
I've thought you wanted to return the error like I've shown you. Did you change your mind or is it an accident?

Reply:
I abandoned this idea as the response of fetch, axios etc are almost always any and this trick does not work :(